s3transfer/download.py

```diff
@@ -171,7 +172,8 @@ def _get_temp_fileobj(self):
         return f
 
     def _get_fileobj(self, filename):
-        f = self._osutil.open(filename, 'wb')
+        f = DeferredOpenFile(filename, mode='wb')
+        f.OPEN_METHOD = self._osutil.open
```

Comment:
Seems reasonable. I can add it. Before, I think i was only using it on tests and the upload method, but since I am doing it twice now, it seems reasonable.